doc/api/addons.md

```diff
@@ -1,4 +1,4 @@
-# Addons
+# C/C++ Addons
```

Comment:
This will require us to also update this internal link at the bottom of the doc?:

<pre>
[Linking to Node.js' own dependencies]: #addons_linking_to_node_js_own_dependencies
</pre>